src/client.ts

```diff
@@ -156,21 +172,62 @@ export class ChattyClient {
             }
             break
           case ChattyHostMessages.MessageWithResponse:
-            const { eventName, payload, sequence } = evt.data.data
-            let results = []
-            if (this._handlers[eventName]) {
-              results = this._handlers[eventName].map(
-                fn => fn.apply(this, payload)
-              )
+            {
+              const { eventName, payload, sequence } = evt.data.data
+              let results = []
+              if (this._handlers[eventName]) {
+                try {
+                  results = this._handlers[eventName].map(
+                    fn => fn.apply(this, payload)
+                  )
+                } catch (error) {
+                  this.sendMsg(ChattyClientMessages.ResponseError, { eventName, payload: error }, sequence)
+                  break
+                }
+              }
+              this.sendMsg(ChattyClientMessages.Response, { eventName, payload: results }, sequence)
+            }
+            break
+          case ChattyHostMessages.MessageWithResponseAsync:
+            {
+              const { eventName, payload, sequence } = evt.data.data
+              let results = []
+              if (this._handlers[eventName]) {
+                results = this._handlers[eventName].map(
+                  fn => fn.apply(this, payload)
+                )
+              }
+              Promise.all(results)
+                .then(resolvedResults => {
+                  this.sendMsg(ChattyClientMessages.Response, { eventName, payload: resolvedResults }, sequence)
+                })
+                .catch(error => {
+                  this.sendMsg(ChattyClientMessages.ResponseError, { eventName, payload: error }, sequence)
+                })
```

Comment:
Since `Promise.all()` accepts both promises and non-promise arguments, this logic doesn't need to be separate and possibly there's not actually a need for two separate messages?